src/rootStore.ts

```diff
@@ -53,8 +53,23 @@ export const storesMap = new WeakMap<
  * Client-side application instance used for devtools
  */
 export let clientApp: App | undefined /*#__PURE__*/
-export const setClientApp = (app: App) => (clientApp = app)
-export const getClientApp = () => clientApp
+export const isClientAppReady = ref(false)
+export const setClientApp = (app: App) => {
+  clientApp = app
+  isClientAppReady.value = true
+}
+let promise: Promise<App> | undefined
+export const getClientApp = () => {
+  promise =
+    promise ||
+    new Promise((resolve) => {
+      watch(
```

Comment:
rather than creating a watcher, I think we should keep the resolve function in a variable that is then called by `setClientApp()`

Reply:
Wow.  I've literally never used that pattern, but that's way easier.  Thanks!